### ISSUE TITLE
svelte props in kebab case syntax not working

### ISSUE THREAD

**Issue by bombillazo** [cloydlau/json-editor-vue#84]
### Before you start, please make sure to:

- [X] Read the document
- [X] Have tried the latest version
- [X] Search to see if your issue already exists
- [X] Look for / ask questions through appropriate channels like [Stack Overflow](https://stackoverflow.com/questions/ask)

### Link to minimal (but complete) reproduction

https://play.vuejs.org/#eNqNUstu2zAQ/JWtLpEBS0rgnlzb6AM+1ECTIgly4kWV1jYdvkBSsgPD/94l5SiNkAY56MDd2dmZ0R6Tb8bkbYPJNJm5ynLjwaFvzIIpLo22Ho5gcQ0nWFst4YKgF31r5bRa1txr+9DgGbCjWoaxmHVgpiqtnIe69CXMA1s6Yoq23HOJuvFpOoL5AtLQz9tSENUcrkZj+l6GHa0UfP30C1XjqJ9yj9LFwSNTQKy+sQpiNXeCV5hejmEy+sLUialZ0XkjV/QgjBGlR3oBzD5lGUxJs0VVo80kLQCtxBPstX10sN+ighrXXGENpQvQ24gMSiDLOpKBa2gzqWsUc5YEVyyB8KSXx4On12AfNV75Y8liVgwo456HmA5XsLq7uZ7C8diFeooWe1vJOOl+UCZLkwce+rsxJXZu0AaaDpVQI/asRslDkSVb742bFkVVK5olE7y1uUJfKCOLZ2gh+J+inRScPBxyuSPCcU9XKi5EmYXFnf6PEL8x1DMOovgA3WDi62V+dVXU3Plhp9MeFlGGJ0rOOzq4Nd8Mcqu0NFygvTGe00G+yq8UQu9XseZtg73saovV4xv1nTt0Fn5bdGhbMtT3fGk3SCcS2su763gufZOOqBFn+/9p3qLTogkaO9j3RtUk+x9cVPszngFXm3u3PHhU7tlUEBrTiPh4Gz/esf4id5J/7lM8/QUknnj7

### Steps to reproduce

Use any of the [svelte-jsoneditor props](https://github.com/josdejong/svelte-jsoneditor/#properties) in kebab vase (the vue convention) and the props will not work with this component.

### What is expected?

To keep the vue convention, any `svelte-jsoneditor` props used in kebab format should work as expected with this component.

### What is actually happening?

One has to use the camelCase prop name with this component, which causes linting issues where the kebab case is enforced.

### System Info

```shell
System:
    OS: macOS 14.4.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 14.44 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.15.1 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.11.0 - /usr/local/bin/npm
    pnpm: 8.15.4 - ~/Library/pnpm/pnpm
    Watchman: 2024.04.01.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 123.0.6312.107
    Safari: 17.4.1
  npmPackages:
    @babel/core: ^7.18.2 => 7.24.4 
    @babel/eslint-parser: ^7.18.2 => 7.24.1 
    @babel/plugin-proposal-class-properties: ^7.18.6 => 7.18.6 
    @babel/plugin-proposal-object-rest-spread: ^7.20.7 => 7.20.7 
    @babel/plugin-proposal-optional-chaining-assign: ^7.24.1 => 7.24.1 
    @babel/plugin-proposal-private-methods: ^7.18.6 => 7.18.6 
    @babel/plugin-proposal-private-property-in-object: ^7.21.11 => 7.21.11 
    @babel/plugin-transform-private-property-in-object: ^7.22.11 => 7.24.1 
    @babel/plugin-transform-runtime: ^7.18.2 => 7.24.3 
    @babel/preset-env: ^7.18.2 => 7.24.4 
    @inquirer/prompts: ^3.2.0 => 3.3.2 
    @rails/actioncable: ^6.1.4-1 => 6.1.7 
    @rails/actiontext: ^6.0.5 => 6.1.7 
    @rails/activestorage: ^7.1.1 => 7.1.3 
    @tabler/icons: ^1.68.0 => 1.119.0 
    @vue/compiler-sfc: ^3.0.5 => 3.4.21 
    axios: ^0.27.2 => 0.27.2 
    babel-loader: ^8.2.5 => 8.3.0 
    babel-plugin-macros: ^3.0.1 => 3.1.0 
    bootstrap: 5.2.1 => 5.2.1 
    chalk: 4.1.2 => 4.1.2 
    chart.js: ^3.8.0 => 3.9.1 
    chartjs-plugin-datalabels: ^2.0.0 => 2.2.0 
    clean-webpack-plugin: ^4.0.0 => 4.0.0 
    color-hash: ^2.0.1 => 2.0.2 
    compression-webpack-plugin: ^10.0.0 => 10.0.0 
    copy-webpack-plugin: ^11.0.0 => 11.0.0 
    core-js: ^3.22.7 => 3.36.1 
    css-loader: ^6.7.1 => 6.11.0 
    cssnano: ^5.1.9 => 5.1.15 
    dompurify: ^2.3.8 => 2.4.9 
    eslint: ^8.16.0 => 8.57.0 
    eslint-config-standard: ^17.0.0 => 17.1.0 
    eslint-plugin-import: ^2.26.0 => 2.29.1 
    eslint-plugin-n: ^15.2.0 => 15.7.0 
    eslint-plugin-node: ^11.1.0 => 11.1.0 
    eslint-plugin-promise: ^6.0.0 => 6.1.1 
    eslint-plugin-vue: ^9.0.1 => 9.24.0 
    inflected: ^2.1.0 => 2.1.0 
    json-editor-vue: ^0.13.0 => 0.13.0 
    leaflet: ^1.8.0 => 1.9.4 
    liquidjs: ^9.37.0 => 9.43.0 
    marked: ^4.0.16 => 4.3.0 
    mini-css-extract-plugin: ^2.6.0 => 2.8.1 
    number-precision: ^1.5.2 => 1.6.0 
    photoswipe: ^5 => 5.4.3 
    postcss: ^8.4.14 => 8.4.38 
    postcss-loader: ^7.0.0 => 7.3.4 
    postcss-preset-env: ^7.6.0 => 7.8.3 
    qs: ^6.10.3 => 6.12.0 
    sass: ^1.69.3 => 1.74.1 
    sass-loader: ^13.3.2 => 13.3.3 
    scroll-behavior-polyfill: ^2.0.13 => 2.0.13 
    stylelint: ^14.8.5 => 14.16.1 
    stylelint-config-sass-guidelines: ^9.0.1 => 9.0.1 
    stylelint-scss: ^4.2.0 => 4.7.0 
    trix: ^1.3.1 => 1.3.1 
    ts-loader: ^9.3.0 => 9.5.1 
    typescript: ^4.7.2 => 4.9.5 
    view3: https://github.com/omohokcoj/view3#motor-admin => 1.0.0 
    vue: ^3.2.0 => 3.4.21 
    vue-draggable-next: ^2.1.1 => 2.2.1 
    vue-loader: ^17.0.0 => 17.4.2 
    vue-router: ^4.0.15 => 4.3.0 
    vue3-ace-editor: 2.2.2 => 2.2.2 
    webpack: ^5.72.1 => 5.91.0 
    webpack-assets-manifest: ^5.1.0 => 5.2.1 
    webpack-bundle-analyzer: ^4.5.0 => 4.10.1 
    webpack-cli: ^4.9.2 => 4.10.0 
    webpack-dev-server: ^4.9.0 => 4.15.2
```

### Package Manager

None

**Comment by cloydlau** [comment]
That's a good point. But `svelte-jsoneditor` does not support props in kebab case too.
You can give it a try here: https://www.sveltelab.dev/q1l38ztdys4at87
For your use case, you can use `v-bind` or turn off this ESLint rule: `'vue/attribute-hyphenation': 'off'`.
Props support automatic casing conversion, but attrs do not, and `'vue/attribute-hyphenation'` cannot distinguish between props and attrs.
One approach is to declare all the properties of `svelte-jsoneditor` in the props of `json-editor-vue` to achieve this, but it's not flexible and too costly.
Another approach is manually converting the casing of attrs, but it fails if `svelte-jsoneditor` also has properties in kebab case.
I think make `'vue/attribute-hyphenation'` only format props (excluding attrs) is maybe the best way.

**Comment by bombillazo** [comment]
Sure, I understand the complexity, but this being a Vue-focused component, it was rather unexpected for it not to follow the Vue convention, especially when linting with recommended Vue rules is involved. I added some exceptions to the rule for hyphenation for the time being, but it would be great to have the consistency of Vue work natively with this component.

It was also very obscured to troubleshoot. Thankfully, I got lucky while editing the component and having auto-refresh on my UI to see the callbacks did not work due to the casing. Still, I recommend a note in the ReadMe to clarify this point since it is not intuitive for Vue users.
